percy/environment.py

```diff
@@ -73,42 +73,52 @@ def target_branch(self):
 
     @property
     def commit_data(self):
-        output = ''
-        if self.commit_sha:
-          output = self._raw_commit_output(self.commit_sha)
-        if not output:
-          output = self._raw_commit_output('HEAD')
+        # Look for commit data set in the environment (usually in the case of being on a CI service)
+        git_data_from_env = {
+            # The only required attribute:
+            'branch': self.branch,
+            # An optional but important attribute:
+            'sha': self.commit_sha,
 
-        # If _raw_commit_output can't be read, return the branch only.
-        if not output:
-          return { 'branch': self.branch }
+            # These GIT_ environment vars are from the Jenkins Git Plugin, but could be
+            # used generically. This behavior may change in the future.
+            'author_name': os.getenv('GIT_AUTHOR_NAME'),
+            'author_email': os.getenv('GIT_AUTHOR_EMAIL'),
+            'committer_name': os.getenv('GIT_COMMITTER_NAME'),
+            'committer_email': os.getenv('GIT_COMMITTER_EMAIL'),
+        }
+
+        git_data_from_git = self._raw_commit_output(self.commit_sha or 'HEAD')
```

Comment:
`self.commit_sha or 'HEAD'` is logically different.  It won't retry the command using `HEAD` if trying the command for `self.commit_sha` fails.